debezium-core/src/main/java/io/debezium/relational/RelationalChangeRecordEmitter.java

```diff
@@ -142,8 +142,9 @@ protected void emitDeleteRecord(Receiver receiver, TableSchema tableSchema) thro
         receiver.changeRecord(getPartition(), tableSchema, Operation.DELETE, oldKey, envelope, getOffset(), null);
     }
 
-    protected void emitTruncateRecord(Receiver receiver, TableSchema schema) throws InterruptedException {
-        throw new UnsupportedOperationException("TRUNCATE not supported");
+    protected void emitTruncateRecord(Receiver receiver, TableSchema tableSchema) throws InterruptedException {
+        Struct envelope = tableSchema.getEnvelopeSchema().truncate(getOffset().getSourceInfo(), getClock().currentTimeAsInstant());
+        receiver.changeRecord(getPartition(), tableSchema, Operation.TRUNCATE, null, envelope, getOffset(), null);
```

Comment:
This logic should actually exist in the Oracle-specific `BaseChangeRecordEmitter` or one of his descendants.  We should leave the exception being thrown in the `RelationalChangeRecordEmitter` since that is shared across all connectors and not all of them support `TRUNCATE` yet.

Reply:
ack, thx.